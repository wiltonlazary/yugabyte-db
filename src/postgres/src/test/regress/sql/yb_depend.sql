-- Save state of pinned objects
CREATE TEMP TABLE pg_depend_copy AS SELECT * FROM pg_depend WHERE deptype = 'p';
CREATE TEMP TABLE pg_shdepend_copy AS SELECT * FROM pg_shdepend WHERE deptype = 'p';
SELECT CASE WHEN count(*) <@ int8range(6000, 6500) THEN 'OK' ELSE 'NOT OK' END FROM pg_depend_copy; -- number of pinned objects between 6000 and 6500, check YBPinnedObjectsCache sizes if fail
SELECT CASE WHEN count(*) <@ int8range(5, 20) THEN 'OK' ELSE 'NOT OK' END FROM pg_shdepend_copy; -- number of pinned shared objects between 5 and 20, check YBPinnedObjectsCache sizes if fail
CREATE TABLE test_table(k INT PRIMARY KEY, v INT);
CREATE VIEW test_view AS SELECT k FROM test_table WHERE v = 10;
CREATE TYPE type_pair AS (f1 INT, f2 INT);
CREATE TYPE type_pair_with_int AS (f1 type_pair, f2 int);
-- Check list of pinned objects has not been changed
SELECT count(*) FROM pg_depend AS d FULL OUTER JOIN pg_depend_copy AS dC ON (d.refclassid = dC.refclassid AND d.refobjid = dC.refobjid) WHERE (dC.refclassid IS NULL AND d.deptype = 'p') OR d.refclassid IS NULL;
SELECT count(*) FROM pg_shdepend AS d FULL OUTER JOIN pg_shdepend_copy AS dC ON (d.refclassid = dC.refclassid AND d.refobjid = dC.refobjid) WHERE (dC.refclassid IS NULL AND d.deptype = 'p') OR d.refclassid IS NULL;
DROP TABLE test_table; -- should fail
DROP TABLE test_table CASCADE;
DROP TYPE type_pair; -- should fail
DROP TYPE type_pair CASCADE;
