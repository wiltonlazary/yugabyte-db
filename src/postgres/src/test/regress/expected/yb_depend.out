-- Save state of pinned objects
CREATE TEMP TABLE pg_depend_copy AS SELECT * FROM pg_depend WHERE deptype = 'p';
CREATE TEMP TABLE pg_shdepend_copy AS SELECT * FROM pg_shdepend WHERE deptype = 'p';
SELECT CASE WHEN count(*) <@ int8range(6000, 6500) THEN 'OK' ELSE 'NOT OK' END FROM pg_depend_copy; -- number of pinned objects between 6000 and 6500, check YBPinnedObjectsCache sizes if fail
 case
------
 OK
(1 row)

SELECT CASE WHEN count(*) <@ int8range(5, 20) THEN 'OK' ELSE 'NOT OK' END FROM pg_shdepend_copy; -- number of pinned shared objects between 5 and 20, check YBPinnedObjectsCache sizes if fail
 case
------
 OK
(1 row)

CREATE TABLE test_table(k INT PRIMARY KEY, v INT);
CREATE VIEW test_view AS SELECT k FROM test_table WHERE v = 10;
CREATE TYPE type_pair AS (f1 INT, f2 INT);
CREATE TYPE type_pair_with_int AS (f1 type_pair, f2 int);
-- Check list of pinned objects has not been changed
SELECT count(*) FROM pg_depend AS d FULL OUTER JOIN pg_depend_copy AS dC ON (d.refclassid = dC.refclassid AND d.refobjid = dC.refobjid) WHERE (dC.refclassid IS NULL AND d.deptype = 'p') OR d.refclassid IS NULL;
 count
-------
     0
(1 row)

SELECT count(*) FROM pg_shdepend AS d FULL OUTER JOIN pg_shdepend_copy AS dC ON (d.refclassid = dC.refclassid AND d.refobjid = dC.refobjid) WHERE (dC.refclassid IS NULL AND d.deptype = 'p') OR d.refclassid IS NULL;
 count
-------
     0
(1 row)

DROP TABLE test_table; -- should fail
ERROR:  cannot drop table test_table because other objects depend on it
DETAIL:  view test_view depends on table test_table
HINT:  Use DROP ... CASCADE to drop the dependent objects too.
DROP TABLE test_table CASCADE;
NOTICE:  drop cascades to view test_view
DROP TYPE type_pair; -- should fail
ERROR:  cannot drop type type_pair because other objects depend on it
DETAIL:  column f1 of composite type type_pair_with_int depends on type type_pair
HINT:  Use DROP ... CASCADE to drop the dependent objects too.
DROP TYPE type_pair CASCADE;
NOTICE:  drop cascades to column f1 of composite type type_pair_with_int
